src/components/canrawsender/canrawsender_p.h

```diff
@@ -1,73 +1,70 @@
 #ifndef CANRAWSENDER_P_H
 #define CANRAWSENDER_P_H
 
-#include <QtGui/QStandardItemModel>
+#include "ui_canrawsender.h"
 #include <QtSerialBus/QCanBusFrame>
-#include <QtWidgets/QHeaderView>
-#include <QtWidgets/QPushButton>
-#include <QtWidgets/QTableView>
-#include <QtWidgets/QToolBar>
-#include <QtWidgets/QVBoxLayout>
-#include <memory>
+#include <QtGui/QStandardItemModel>
 
-class CanRawSenderPrivate : public QObject {
+namespace Ui {
+class CanRawSenderPrivate;
+}
+
+class CanRawSenderPrivate : public QWidget {
     Q_OBJECT
     Q_DECLARE_PUBLIC(CanRawSender)
 
 public:
     CanRawSenderPrivate(CanRawSender* q)
-        : tvModel(0, 3)
-        , pbAdd("Add")
-        , q_ptr(q)
+    : ui(new Ui::CanRawSenderPrivate)
+    , q_ptr(q)
     {
+        ui->setupUi(this);
+
+        tvModel.setHorizontalHeaderLabels({"id", "value",""});
+        ui->tv->setModel(&tvModel);
+
+        connect(ui->pbAdd, &QPushButton::pressed, this, &CanRawSenderPrivate::addNewItem);
     }
 
-    void setupUi()
+    ~CanRawSenderPrivate()
     {
-        tvModel.setHorizontalHeaderLabels({ tr("id"), tr("data"), "" });
-        tv.setModel(&tvModel);
-        tv.verticalHeader()->hideSection(0);
+        delete ui;
+    }
 
-        toolbar.addWidget(&pbAdd);
+private:
+    Ui::CanRawSenderPrivate* ui;
```

Comment:
I think this can be converted to std::unique_ptr. Delete in dtor will not be required then.